### PR TITLE
2048 aytq migrate domains

### PR DESCRIPTION
**Pull request by shaheislamdfe**
### Context

Migration of frontdoor and domains infra to s189.

### Changes proposed in this pull request

Addition of domain infrastructure and environment domain references in Terraform, aswell as makefile updates with domains-infra and domains targets.

### Guidance to review

Validate front door and domains/hosted zones sit within s189 via UI and domains are healthy.

https://access-your-teaching-qualifications.education.gov.uk/
https://check-a-teachers-record.education.gov.uk/

### Link to Trello card

https://trello.com/c/kYWAOZTW/2048-aytq-migrate-domains

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by shaheislamdfe** [comment]
[AYTQ: migrate domains](https://trello.com/c/kYWAOZTW/2048-aytq-migrate-domains)

File: terraform/domains/environment_domains/config/development.tfvars.json
Comment (saliceti) [inline terraform/domains/environment_domains/config/development.tfvars.json:1]: This file can be removed as we don't use the dev environment

File: terraform/domains/environment_domains/config/development_Terrafile
Comment (saliceti) [inline terraform/domains/environment_domains/config/development_Terrafile:1]: This file can be removed as we don't use the dev environment

File: terraform/domains/environment_domains/config/preprod_Terrafile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+domains:
+    source:  "https://github.com/DFE-Digital/terraform-modules"
+    version: "stable"
```
Comment (saliceti) [inline terraform/domains/environment_domains/config/preprod_Terrafile]: Could we use testing to catch potential errors before production

**Review by saliceti — APPROVED** [review]

**Pull request merged**